mypy/build.py

```diff
@@ -2695,13 +2708,20 @@ def load_graph(sources: List[BuildSource], manager: BuildManager,
         #   (since direct dependencies reflect the imports found in the source)
         #   but A's cached *indirect* dependency on C is wrong.
         dependencies = [dep for dep in st.dependencies if st.priorities.get(dep) != PRI_INDIRECT]
-        added = [dep for dep in st.suppressed
-                 if manager.find_module_cache.find_module(dep, manager.search_paths,
-                                                          manager.options.python_executable)]
+        if not manager.use_fine_grained_cache():
```

Comment:
Is this change necessary to fix the performance regression, or is this just a somewhat related cleanup? If this is not required, I'd propose to move this into a separate PR, since the import suppression logic is very delicate and we should do whatever we can to make reasoning about changes easier.

Reply:
Performance. Cloning options for a lot of files may take time. For example in S repo this could be 5-10 sec, which is negligible for normal incremental run, but is significant for daemon cold start.

Reply:
Hm, I put a cache in for that long ago. Did the rewrite for mypyc break that again?